.github/workflows/tests.yml

```diff
@@ -0,0 +1,169 @@
+          echo "test_functions=${test_functions}"
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    needs:
+      - configure-tests
+    strategy:
+      fail-fast: false # don't propagate failing jobs
+      matrix:
+        test_function: ${{ fromJson(needs.configure-tests.outputs.test_functions) }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Install tooling using asdf
+        uses: asdf-vm/actions/install@05e0d2ed97b598bfce82fd30daf324ae0c4570e6 # v3
+
+      - name: Import Secrets
+        id: secrets
+        uses: hashicorp/vault-action@d1720f055e0635fd932a1d2a48f87a666a57906c # v3
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          method: approle
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+          exportEnv: false
+          secrets: |
+            secret/data/products/infrastructure-experience/ci/common AWS_ACCESS_KEY;
+            secret/data/products/infrastructure-experience/ci/common AWS_SECRET_KEY;
+
+      # Official action does not support profiles
+      - name: Add profile credentials to ~/.aws/credentials
+        run: |
+          aws configure set aws_access_key_id ${{ steps.secrets.outputs.AWS_ACCESS_KEY }} --profile ${{ env.AWS_PROFILE }}
+          aws configure set aws_secret_access_key ${{ steps.secrets.outputs.AWS_SECRET_KEY }} --profile ${{ env.AWS_PROFILE }}
+          aws configure set region ${{ env.AWS_REGION }} --profile ${{ env.AWS_PROFILE }}
+
+      - name: Get go.mod details
+        uses: Eun/go-mod-details@b719cd324463e2037cf3a0dd1dd6091bdc2730f4 # v1
+        id: go-mod-details
+        with:
+          modfile: ${{ github.workspace }}/test/src/go.mod
+
+      - name: Launch test
+        timeout-minutes: 125
+        run: |
+          export TESTS_CLUSTER_ID="${{ needs.configure-tests.outputs.cluster_id }}"
+          export TESTS_CLUSTER_REGION="${{ env.AWS_REGION }}"
+          export TESTS_DISABLE_KMS_CHECKS=true
+          export TESTS_TF_BINARY_NAME="${{ env.TESTS_TF_BINARY_NAME }}"
+          just test ${{ matrix.test_function }} "--junitfile ${{ matrix.test_function }}_unit-tests.xml"
+
+      # this is a workaround for test report not working as expected due to https://github.com/test-summary/action/issues/5
+      - name: Filter logger.go from the test report (too large)
+        if: always()
+        run: |
+          sed 's/&#xA;/\n/g' < "./test/src/${{ matrix.test_function }}_unit-tests.xml" | grep -E -v '^.*logger\.go.*$' | sed 's/\n/&#xA;/g' > "./test/src/${{ matrix.test_function }}_unit-tests_filtered.xml"
+
+      - name: Upload test reports
+        if: always()
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3
+        with:
+          name: test-reports
+          path: "./test/src/${{ matrix.test_function }}_unit-tests_filtered.xml"
```

Comment:
just a suggestion to define retention days, not sure we need them by default for 90 days?